### PR TITLE
Maximum Function

### LINKED ISSUE #3519: Maximum Function
## 🚀 Feature

Maximum Function (in Java) : 

A sequence is given A1,A2,…,An . Find the maximum value of the expression |Ax−Ay|+|Ay−Az|+|Az−Ax| over all triples of pairwise distinct valid indices (x,y,z).

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

One of the important Competitive programming question

## Pitch

Important for beginners

### PR DESCRIPTION
**Pull request by Amanduttbhagat**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

  Maximum Function (in Java) :
 
   A sequence is given A1,A2,…,An (atleast 3 elements you need to give)
   Find the maximum value of the expression |Ax − Ay|+|Ay − Az|+|Az − Ax|
   over all triples of pairwise distinct valid indices (x,y,z).

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

fixed : #3519

### CODE REVIEW

**Comment by Amanduttbhagat** [comment]
@ankitaggarwal23 @HarshCasper @siddharth25pandey @iamrajiv
Please review my PR

File: Java/cp/Maximum_Function.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+*/
+import java.util.Arrays;
+import java.util.Scanner;
+
+class MaximumFunction {
+	//main method
+	public static void main(String[] args) {
+
+		Scanner sc = new Scanner(System.in);
+
```
Comment (siddharth25pandey) [inline Java/cp/Maximum_Function.java:22]: Display message, what are you taking input from the user

File: Java/cp/Maximum_Function.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		long n = sc.nextLong();
+		long[] array = new long[(int)n];
+		//putting elements in the array
```
Comment (siddharth25pandey) [inline Java/cp/Maximum_Function.java:25]: Display message, what are you taking input from the user

File: Java/cp/Maximum_Function.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		for(long t = 0 ; t<n ; t++) {
+			array[(int)t] = sc.nextLong();
+		}
+		//sorting the array
+		Arrays.sort(array);
+		//calculating max
+		long max = Math.abs(array[0]-array[1]) + Math.abs(array[1]-array[(int)n-1]) + Math.abs(array[(int)n-1]-array[0]);
+		//printing max
```
Comment (siddharth25pandey) [inline Java/cp/Maximum_Function.java:33]: Display output message for better user-interface

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@Amanduttbhagat Do the suggested changes

File: Java/cp/Maximum_Function.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+   over all triples of pairwise distinct valid indices (x,y,z).
+
+Approach :
+  1. We need to sort the given array
+  2. To find maximum difference of two elements we will subtract 1st element with the last element of the array (|Ax − Ay|)
+  3. Next, we can take middle element as 2nd element of the array (Az)
+  4. That means, Ax = 1st element of the sorted array, Ay = last element, Az = 2nd element
+  5. Therefore, adding |Ax − Ay|+|Ay − Az|+|Az − Ax| will give maximum result everytime
+*/
+import java.util.Arrays;
```
Comment (ankitaggarwal23) [inline Java/cp/Maximum_Function.java:15]: Add a new line after description

File: Java/cp/Maximum_Function.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.Scanner;
+
+class MaximumFunction {
+	//main method
+	public static void main(String[] args) {
+
+		Scanner sc = new Scanner(System.in);
+
+		long n = sc.nextLong();
+		long[] array = new long[(int)n];
```
Comment (ankitaggarwal23) [inline Java/cp/Maximum_Function.java]: Display a proper message How you are reading inputs .

File: Java/cp/Maximum_Function.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+}
+/*
+ - Test Cases :
+   1.  Input : 5
+               3 4 6 7 1
+      Output : Result = 12
+
+   2.  Input : 10
+               7 5 3 9 1 2 7 3 8 9
+      Output : Result = 16
```
Comment (ankitaggarwal23) [inline Java/cp/Maximum_Function.java]: Remove Extra test case one is enough

File: Java/cp/Maximum_Function.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+ - Complexity Analysis :
+   * Time Complexity : O(n)
+       We are using only one for loop, so it iterates maximum 'n' times while traversing for loop
+
+   * Space Complexity : O(n)
+       Auxiliary space : O(1)  +  input space : O(n)
+ */
```
Comment (ankitaggarwal23) [inline Java/cp/Maximum_Function.java:58]: Add a new line at the end of file

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Amanduttbhagat** [comment]
@ankitaggarwal23 I have made the requested changes
Can you please review it again

Thank you

**Comment by siddharth25pandey** [comment]
@Amanduttbhagat Displaying message means you need to print the message not writing as in comment form

**Comment by Amanduttbhagat** [comment]
@ankitaggarwal23 @siddharth25pandey
I have made the requested changes
Please review it again

Thank you

**Review by HarshCasper — APPROVED** [review]

**Comment by Amanduttbhagat** [comment]
Thank you @HarshCasper @ankitaggarwal23 @siddharth25pandey

**Comment by siddharth25pandey** [comment]
@all-contributors please add @Amanduttbhagat for his code

**Pull request merged**